### DISCUSSION TITLE
Route by domain without fakedns

### DISCUSSION THREAD

**Discussion by Temtaime** [XTLS/Xray-core#4202] [Q&A]
Hello !
I am trying to set up a routing based on domain name.
With fakedns i can route connections with `sniffing: fakedns`.
What i am trying to do is to route some connections based on domain name. It is not a http(s) traffic, so it cannot be sniffed.
Fakedns works great, but i don't want to fakedns because some resources start to work incorrectly.
Do xray provide some kind of transparent dns proxy used in routing ?
Thanks

**Comment by nomadacipala-ops** [comment]
You can route by domain without FakeDNS by using SNI inspection, reverse proxies, or proper DNS-based routing at the server level. While experimenting with multi-domain setups on domainoffer, I found reverse proxy rules much cleaner and easier to maintain long term.
